src/Core/Worker.php

```diff
@@ -395,24 +395,24 @@ private static function execFunction($queue, $funcname, $argv, $method_call)
 			$a->callstack = [];
 		}
 
-		// For better logging create a new process id for every worker call
-		// But preserve the old one for the worker
-		$old_process_id = $a->process_id;
-		$a->process_id = $new_process_id;
-		$a->queue = $queue;
-
 		$up_duration = microtime(true) - self::$up_start;
 
 		// Reset global data to avoid interferences
 		unset($_SESSION);
 
 		if ($method_call) {
-			call_user_func_array(sprintf('Friendica\Worker\%s::execute', $funcname), $argv);
+			$oldLogger = Logger::switchLogger($workerLogger);
+			$worker = WorkerFactory::create($funcname, $a, $workerLogger);
+			/**
+			 * This call uses the array as a parameter because of this benchmark
+			 * https://github.com/fab2s/call_user_func/
```

Comment:
Please move this comment to the commit/PR, as it only makes sense in the context of this diff with the previous version.